core/src/main/scala/filodb.core/query/QueryUtils.scala

```diff
@@ -0,0 +1,25 @@
+package filodb.core.query
+
+/**
+ * Storage for utility functions.
+ */
+object QueryUtils {
+  private val regexChars = Array('.', '?', '+', '*', '|', '{', '}', '[', ']', '(', ')', '"', '\\')
+  private val regexCharsMinusPipe = (regexChars.toSet - '|').toArray
+
+  /**
+   * Returns true iff the argument string contains any special regex chars.
+   */
+  def containsRegexChars(str: String): Boolean = {
+    str.exists(regexChars.contains(_))
+  }
+
+  /**
+   * Returns true iff the argument string contains no special regex
+   *   characters except the pipe character ('|').
+   * True is also returned when the string contains no pipes.
+   */
+  def isPipeOnlyRegex(str: String): Boolean = {
+    str.forall(!regexCharsMinusPipe.contains(_))
```

Comment:
I've (more-or-less) copy-pasted the Lucene logic here to keep the diff simple/safe. To correctly handle escaped characters, we'd need something 10x less pretty:
```
  /**
   * Returns the set of unescaped special regex chars in the argument string.
   * Special chars are: . ? + * | { } [ ] ( ) " \
   */
  def getUnescapedSpecialRegexChars(str: String): Set[Char] = {
    // Match special chars preceded by any count of backslash pairs and either
    //   some non-backslash character or the beginning of a line.
    val regex = "(?<=(^|[^\\\\]))(\\\\\\\\)*([.?+*|{}\\[\\]()\"\\\\])".r
    regex.findAllMatchIn(str)
      .map(_.group(3)) // get the special char -- third capture group
      .map(_(0)) // convert the string to a char
      .toSet
  }
```